### PR TITLE
79 add text node

### LINKED ISSUE #79: Create Text Nodes
Add new option for "Media Type" called Text

Choosing this option will present the user with a simple textarea where they can type in text and save.

The tapestry should display this like a regular node and show the text in a pop-up similar to videos / h5p with formatting so it's readable.

### PR DESCRIPTION
**Pull request by kevinthewebguy**
fixes #79

### CODE REVIEW

File: tapestry.js [hunk 1]
```diff
@@ -2026,6 +2164,37 @@ function exitViewMode() {
...
+
+    var titleText = document.createElement("p");
+    titleText.appendChild(document.createTextNode(title));
+    titleSection.append(titleText);
+    lightboxContent.append(titleSection);
+
+    if (str) {
+        var paragraphSection = document.createElement("div");
+        paragraphSection.setAttribute("id", "text-light-box-paragraph");
+        paragraphArray = str.split("\n");
```
Comment (wynnset) [inline tapestry.js]: Can you please change this to `\n\n`? And then in the body of the iteration, replace any single instances of `\n` with `<br>`. That way, people can still add single line breaks.

File: tapestry.js [hunk 2]
```diff
@@ -2026,6 +2164,37 @@ function exitViewMode() {
...
+        for (var i = 0; i < paragraphArray.length; i++) {
+            if (paragraphArray[i] !== "") {
```
Comment (wynnset) [inline tapestry.js]: Let's remove this as users may be adding in a blank line on purpose.

**Comment by wynnset** [comment]
Hey @kevinthewebguy I made some changes to improve the UI a bit and I also fixed some bugs that I found. Please do a pull before you start working and then address the comments I left you above. Thanks!

File: tapestry.js [hunk 3]
```diff
@@ -1416,6 +1489,46 @@ function buildPathAndButton() {
         recordAnalyticsEvent('user', 'open', 'lightbox', thisBtn.dataset.id);
     });
 
+    // Append text node button
+    nodes
```
[the remark this answers is not in the document]
Comment (kevinthewebguy) [reply tapestry.js]: Used the same code and used mediatype to filter.

File: tapestry.js [hunk 4]
```diff
@@ -1472,12 +1585,15 @@ function buildPathAndButton() {
...
 
-    $('.addNodeButton > i').click(function(){
+    $('.addNodeButton').click(function(){
         // Set up the title of the form
         $('#createNewNodeModalLabel').text("Add new sub-topic to " + dataset.nodes[findNodeIndex(root)].title);
         $("#submit-add-root-node").hide();
         $("#submit-edit-node").hide();
         $("#submit-add-new-node").show();
+        if (dataset.nodes[findNodeIndex(root)].mediaType !== "video") {
+            $("#appearsat-section").hide();
```
[the remark this answers is not in the document]
Comment (kevinthewebguy) [reply tapestry.js]: Looks like now the unlocked checkbox shows up for children of text nodes.

File: tapestry.js (hunk 2 above)
Comment (kevinthewebguy) [reply tapestry.js]: Will add line break on else statement

File: tapestry.js (hunk 1 above)
Comment (kevinthewebguy) [reply tapestry.js]: Modified so that user can add line breaks

**Pull request merged**